src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -189,7 +201,7 @@ public IndexShard shardSafe(int shardId) throws IndexShardMissingException {
         return indexShard;
     }
 
-    public ImmutableSet<Integer> shardIds() {
+    public Set<Integer> shardIds() {
```

Comment:
Same here, I think we should go back to the `ImmutableSet`, the types should keep people from thinking they can change things with these collections